src/metaUtils.cxx

```diff
@@ -466,63 +467,126 @@ MET_ValueToValue(MET_ValueEnumType _fromType,
+    {
+      auto ptr = static_cast<MET_LONG_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_LONG_TYPE>(tf);
       return true;
+    }
     case MET_UINT:
     case MET_UINT_ARRAY:
-      ((static_cast<MET_UINT_TYPE *>(_toData))[_index]) = static_cast<MET_UINT_TYPE>(tf);
+    {
+      auto ptr = static_cast<MET_UINT_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_UINT_TYPE>(tf);
       return true;
+    }
     case MET_ULONG:
     case MET_ULONG_ARRAY:
-      ((static_cast<MET_ULONG_TYPE *>(_toData))[_index]) = static_cast<MET_ULONG_TYPE>(tf);
+    {
+      auto ptr = static_cast<MET_ULONG_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_ULONG_TYPE>(tf);
       return true;
+    }
     case MET_LONG_LONG:
     case MET_LONG_LONG_ARRAY:
-      ((static_cast<MET_LONG_LONG_TYPE *>(_toData))[_index]) = static_cast<MET_LONG_LONG_TYPE>(tf);
+    {
+      auto ptr = static_cast<MET_LONG_LONG_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_LONG_LONG_TYPE>(tf);
       return true;
+    }
     case MET_ULONG_LONG:
     case MET_ULONG_LONG_ARRAY:
-      ((static_cast<MET_ULONG_LONG_TYPE *>(_toData))[_index]) = static_cast<MET_ULONG_LONG_TYPE>(tf);
+    {
+      auto ptr = static_cast<MET_ULONG_LONG_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_ULONG_LONG_TYPE>(tf);
       return true;
+    }
     case MET_DOUBLE:
     case MET_DOUBLE_ARRAY:
-      ((static_cast<MET_DOUBLE_TYPE *>(_toData))[_index]) = static_cast<MET_DOUBLE_TYPE>(tf);
+    {
+      auto ptr = static_cast<MET_DOUBLE_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_DOUBLE_TYPE>(tf);
       return true;
+    }
     case MET_FLOAT:
     case MET_FLOAT_ARRAY:
     case MET_FLOAT_MATRIX:
-      ((static_cast<MET_FLOAT_TYPE *>(_toData))[_index]) = static_cast<MET_FLOAT_TYPE>(tf);
+    {
+      auto ptr = static_cast<MET_FLOAT_TYPE *>(_toData);
+      ptr[_index] = static_cast<MET_FLOAT_TYPE>(tf);
       return true;
+    }
     case MET_STRING:
-      sprintf(&((static_cast<MET_ASCII_CHAR_TYPE *>(_toData))[_index]), "%f", tf);
+    {
+      auto ptr = static_cast<MET_ASCII_CHAR_TYPE *>(_toData);
+      snprintf(&(ptr[_index]), _toDataSize - _index, "%f", tf);
       return true;
+    }
     case MET_NONE:
     case MET_OTHER:
     default:
       return false;
   }
 }
 
+bool
+MET_ValueToValue(MET_ValueEnumType _fromType,
+                 const void *      _fromData,
+                 std::streamoff    _index,
+                 MET_ValueEnumType _toType,
+                 void *            _toData,
+                 double            _fromMin,
+                 double            _fromMax,
+                 double            _toMin,
+                 double            _toMax)
+{
+  // We don't know the buffer size, so just use a big number.
+  return MET_ValueToValue(_fromType,
+                          _fromData,
+                          _index,
+                          _toType,
+                          _toData,
+                          INT_MAX,
```

Comment:
This is only used for strings, not pixel buffers, right? Otherwise this will have a 2GB limit on Windows.

Reply:
Hmm, I'll switch to SIZE_MAX...

Reply:
Done.